### DISCUSSION TITLE
Would it be possible to support sections that does not listen to precedence in the closer EditorConfig file?

### DISCUSSION THREAD

**Discussion by xuhdev** [editorconfig/editorconfig#525] [General]
Would it be possible to support sections that does not listen to precedence in the closer EditorConfig file?

Given an example:
I have a file: "\~/.editorconfig" that mentions the following:
```
# top-most EditorConfig file
root = true

[*]
charset = utf-8
trim_trailing_whitespace = true
insert_final_newline = true

[{.git**,*.pfx}]
charset = unset
end_of_line = unset
insert_final_newline = unset
trim_trailing_whitespace = unset
indent_style = unset
indent_size = unset
```

Then, in a repository that I use to collaborate with others, we have the following file: "\~/projects/my-collaborative-project/.editorconfig" that has the following:
```
[*]
charset = utf-8
trim_trailing_whitespace = true
insert_final_newline = true

# Should match "go fmt"
[*{.go}]
indent_style = tab
indent_size = 4
```

I really want the settings in "\~/.editorconfig" regarding files with ".git/\*" and "\*.pfx" to take effect, since my IDE is complaining about .editorconfig violations in these files.
This is of course due to my "\~/projects/my-collaborative-project/.editorconfig" saying that I have a "*" defined and set rules for every file. But I want a certain section in the root to take precedence.

Is this something I could create an issue requesting this functionality?

Perhaps a property that is called "precedence" that could be set to "use_parent_setting"?

Or perhaps a way to specify files not previously defined, like so:
```
[%*] # "%*" means apply to any file unless other rules specify other properties.
charset = utf-8
trim_trailing_whitespace = true
insert_final_newline = true
```

And of course also document the feature in [File Processing](https://spec.editorconfig.org/#id10), specifically: [...most recent rules found take precedence...](https://spec.editorconfig.org/#:~:text=files%20are%20read%20top%20to%20bottom%20and%20the%20most%20recent%20rules%20found%20take%20precedence.%20if%20multiple%20editorconfig%20files%20have%20matching%20sections%2C%20the%20rules%20from%20the%20closer%20editorconfig%20file%20are%20read%20last%2C%20so%20pairs%20in%20closer%20files%20take%20precedence)

Sorry to piggieback off an old issue, but the fix in https://github.com/editorconfig/editorconfig/commit/4f21a9c127e801be2bbd20600efb06e9472d2d90 is the one causing headaches for me.

_Originally posted by @tonur in https://github.com/editorconfig/editorconfig/issues/14#issuecomment-2451663969_

**Comment by xuhdev** [comment]
@tonur 

You can append the following to the project `.editorconfig`:

```
[.git/*]
charset = unset
trim_trailing_whitespace = unset
insert_final_newline = unset
```

Or, you can add the following to the `.git` dir:

```
root = true
```

(Your project editorconfig file should contain `root = true`, otherwise your IDE may behave inconsistently from your teammates' IDE due to the presence of `~/.editorconfig`.)

**Comment by cxw42** [comment]
> Your project editorconfig file should contain root = true,

I agree that this is the right approach here.

Applying settings from outside a project (e.g., your `~/.editorconfig` file rules for `*.pfx`) has been requested at #236 .
